cnd/src/network/protocol.rs

```diff
@@ -0,0 +1,162 @@
+use crate::swap_protocols::{rfc003::SecretHash, SwapId};
+use futures::{AsyncRead, AsyncWrite, Future};
+use libp2p::core::{upgrade, InboundUpgrade, OutboundUpgrade, UpgradeInfo};
+use serde::{Deserialize, Serialize};
+use std::{iter, pin::Pin};
+use tracing::trace;
+
+/// The secret hash sharing protocol works in the following way:
+///
+/// - Dialer (Alice) writes the `Message` to the substream.
+/// - Listener (Bob) reads the `Message` from the substream.
+
+/// Data sent to peer in secret hash protocol.
+#[derive(Clone, Copy, Deserialize, Debug, Serialize, PartialEq)]
+pub struct Message {
+    swap_id: SwapId,
+    secret_hash: SecretHash,
+}
+
+/// Represents a prototype for an upgrade to handle the sender side of the
+/// secret hash sharing protocol.  Config contains the `SecretHash` so that once
+/// the outbound upgrade is complete peer node has been sent the secret hash.
+#[derive(Clone, Copy, Debug)]
+pub struct OutboundProtocolConfig {
+    msg: Message,
+}
+
+impl UpgradeInfo for OutboundProtocolConfig {
+    type Info = &'static [u8];
+    type InfoIter = iter::Once<Self::Info>;
+
+    fn protocol_info(&self) -> Self::InfoIter {
+        iter::once(b"/comit/swap/secret_hash/1.0.0")
```

Comment:
I think it would be good to share this protocol string in a const between the two configs.

Reply:
Lols, see this comment above: https://github.com/comit-network/comit-rs/pull/2215#discussion_r390159147